crates/nu-protocol/src/eval_const.rs

```diff
@@ -510,6 +510,19 @@ impl Eval for EvalConst {
     ) -> Result<Value, ShellError> {
         // TODO: Allow debugging const eval
         let block = working_set.get_block(block_id);
+
+        // If parsing errors exist in the block, don't bother to evaluate it.
+        if let Some(block_span) = block.span {
+            if working_set
+                .parse_errors
+                .iter()
+                .any(|error| block_span.contains_span(error.span()))
+            {
+                return Err(ShellError::NotAConstant { span });
+            }
+        } else if !working_set.parse_errors.is_empty() {
+            return Err(ShellError::NotAConstant { span });
```

Comment:
would these errors ever be shown to the user? if yes, could we potentially turn the parse error into a shell error via a LabeledError, like:
```rust
ShellError::LabeledError(Box::new(err.clone().into()))
```

I don't think there's precedent for this, but I ran into a similar situation recently. I would like to get @sholderbach's opinion on this before going that route as well.

if these wouldn't ever be shown to the user, maybe this should be a `ShellError::NushellFailed` instead?

Reply:
I've no idea of how to create a block with span set to None.

Reply:
But the error in the first branch is definitely shown to the user:

<img width="494" alt="image" src="https://github.com/user-attachments/assets/ea8cdbfb-61c5-42f4-aec3-fc61a089412c" />

Reply:
<img width="641" alt="image" src="https://github.com/user-attachments/assets/41fe5616-7058-4310-87d6-23ca09183b90" />

Better to get the parser error's message to convert to ShellError. But the trait bound for converting parser error to shell error is not implemented yet, have to create one from scratch.

Reply:
Ended up with a new type of ShellError. Please review it again.